lib/_http_agent.js

```diff
@@ -115,6 +115,9 @@ Agent.prototype.addRequest = function(req, options) {
     };
   }
 
+  options = util._extend({}, options);
```

Comment:
This can be done as a single call like `options = Object.assign({}, options, this.options);`